sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyRotationPolicy.java

```diff
@@ -42,54 +37,100 @@ public String getId() {
     }
 
     /**
-     * Get the {@link KeyRotationPolicy policy's} created time in UTC.
+     * Get the actions that will be performed by Key Vault over the lifetime of a key.
      *
-     * <p>May be undefined if a {@link KeyRotationPolicy policy} has not been explicitly set.</p>
+     * <p>You may also pass an empty array to restore to its default values.</p>
      *
-     * @return The {@link KeyRotationPolicy policy's} created time in UTC.
+     * @return The {@link KeyRotationLifetimeAction actions} in this {@link KeyRotationPolicy policy}.
      */
-    public OffsetDateTime getCreatedOn() {
-        return this.createdOn;
+    public List<KeyRotationLifetimeAction> getLifetimeActions() {
+        return this.lifetimeActions;
     }
 
     /**
-     * Get the {@link KeyRotationPolicy policy's} last updated time in UTC.
+     * Set the actions that will be performed by Key Vault over the lifetime of a key.
      *
-     * <p>May be undefined if a {@link KeyRotationPolicy policy} has not been explicitly set.</p>
+     * <p>You may also pass an empty array to restore to its default values.</p>
      *
-     * @return The {@link KeyRotationPolicy policy's} last updated time in UTC.
+     * @param lifetimeActions The {@link KeyRotationLifetimeAction actions} to set.
+     *
+     * @return The updated {@link KeyRotationPolicy} object.
      */
-    public OffsetDateTime getUpdatedOn() {
-        return this.updatedOn;
+    public KeyRotationPolicy setLifetimeActions(List<KeyRotationLifetimeAction> lifetimeActions) {
+        this.lifetimeActions = lifetimeActions;
+
+        return this;
     }
 
     /**
-     * Set the optional key expiration period used to define the duration after which a newly rotated key will expire.
+     * Get the optional key expiration period used to define the duration after which a newly rotated key will expire.
      * It should be defined as an ISO 8601 duration. For example, 90 days would be formatted as follows: "P90D", 3
-     * months would be "P3M", 48 hours would be "PT48H" and 1 year and 10 days would be "P1Y10D".
+     * months would be "P3M" and 1 year and 10 days would be "P1Y10D".
      *
-     * @param expiryTime The expiry time to set in ISO 8601 format.
+     * @return The expiration time in ISO 8601 format.
+     */
+    public String getExpiresIn() {
```

Comment:
We should return `Duration` instead of `String` here.

Reply:
We had a long discussion on this a few months back (maybe when @srnagar was on leave). My recollection was that @vcolin7 was going to look into options, including `Duration`, `Period`, etc, as well as rolling our own solution. I can't recall seeing the results of this investigation, but I would like to be sure that we can't (or shouldn't) do a more higher-level concept than String.

Reply:
I'm starting to remember the issues @vcolin7 found. `Duration` is good for measuring time in terms of seconds and nanoseconds, and `Period` is good for measuring time in terms of days, months, and years. From this I believe it was determined that neither of these appropriately modelled the possible values that users could pass in to this service, but I would like to see confirmation from @vcolin7 on that.

Reply:
We should look into how this will be used by customers as I am guessing customers will have similar issues using a `String` type. Having a good sample of a customer scenario will demonstrate the complexity the customer will have to deal with. If it's straightforward to use, then we can leave it as string but if we find that it's not trivial, we may want to introduce a type that supports all values. 

For reference, you can see [PeriodDuration](https://www.threeten.org/threeten-extra/apidocs/org.threeten.extra/org/threeten/extra/PeriodDuration.html) that attempts to solve this.

Reply:
Had a separate discussion with @srnagar and @heaths and, since these values will be originally provided by users, are round-trippable and won't normally need to be manipulated, we came to the conclusion that we will keep the following values as `String`:

- `expiresIn` in `KeyRotationPolicy`
- `timeBeforeExpiry` and `timeAfterCreate` in `KeyRotationPolicyLifetimeActions`

We will, however, add sufficient documentation in our code and samples to allow users to provide properly formed ISO 8601 duration values.

Reply:
I agree a `String` for now is probably the best option. None of the languages we currently support have support for ISO 8601 durations out of the box and I think, given their growing ubiquity, we should consider adding them in Core. But we likely need to consider all the nuances like fractions in the last digit, and go through customer studies.